OpenRA.Mods.Common/Traits/Health.cs

```diff
@@ -168,11 +168,22 @@ public void InflictDamage(Actor self, Actor attacker, Damage damage, bool ignore
 			// Apply any damage modifiers
 			if (!ignoreModifiers && damage.Value > 0)
 			{
-				var modifiers = damageModifiers
-					.Concat(damageModifiersPlayer)
-					.Select(t => t.GetDamageModifier(attacker, damage));
+				var modifiers = new List<int>();
+				foreach (var dm in damageModifiers)
```

Comment:
I'm surprised that adding these extra loops and then looping over the results again to convert the list to an array isn't slower. What if we check for 100 and skip inside `ApplyPercentageModifiers` instead?

Reply:
> What if we check for 100 and skip inside ApplyPercentageModifiers instead?

I tried, there was next to no win. Haven't profiled the #6079 stuff yet, but I tried using only normal integer math instead of decimal, still not much of a difference.

> I'm surprised that adding these extra loops and then looping over the results again to convert the list to an array isn't slower.

That's what stop-watching it showed.
For the record, I did do some back-and-forth testing, as well as separately profiling the `var` creation and application of modifiers to ensure it wasn't just a fluke, forgotten compile or anything like that.